app/src/main/java/net/squanchy/navigation/HomeActivity.java

```diff
@@ -49,4 +80,58 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                 }
         );
     }
+
+    private void selectPage(BottomNavigationSection section) {
+        if (section == currentSection) {
+            return;
+        }
+
+        Fade transition = new Fade();
+        TransitionManager.beginDelayedTransition(pageContainer, transition);
+
+        if (currentSection != null) {
+            pageViews.get(currentSection).setVisibility(View.INVISIBLE);
+        }
+        pageViews.get(section).setVisibility(View.VISIBLE);
+
+        Resources.Theme theme = getThemeFor(section);
+        setStatusBarColor(getColorFromTheme(theme, android.R.attr.statusBarColor));
+        setBottomNavigationBarColor(getColorFromTheme(theme, android.support.design.R.attr.colorPrimary));
+
+        currentSection = section;
+    }
+
+    private Resources.Theme getThemeFor(BottomNavigationSection section) {
+        Resources.Theme theme = getResources().newTheme();
+        theme.setTo(getTheme());
+        theme.applyStyle(section.theme(), true);
```

Comment:
`setTo()` basically is a "copy everything from". `applyStyle(themeId, true)` means "apply all the attributes from the `themeId` into `theme`, overriding anything that might have been defined there already. It's basically the same that `ContextThemeWrapper` does internally.